components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -3719,7 +3715,6 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
 
     protected censorUser(user: User): User {
         const res = { ...user };
-        delete res.additionalData;
```

Comment:
@AlexTugarev Did you check if this was there for a reason? I believe we added it because we're sending out secrets/tokens out with that one, no?

Reply:
In general, the "censor" functions are there to strip sensitive info from shapes before serialization, e.g. cmp censorWorkspace. So if we're using this one differently, I'd be in favor of a different name. :+1:

Reply:
(also, in the past we relied on re-retrieving the User shape in DB in places. But I agree that we don't need that additional roundtrip :+1: )

Reply:
@geropl, in fact removing `additionalData` from being sent to the frontend client seems to be a bug. 

See, this bad of misc user data is free to change for you. The rule is implemented [here](https://github.com/gitpod-io/gitpod/blob/35f3c98b980991d15cdfe30d5b82323efc0dd52d/components/server/src/user/user-service.ts#L409). 

I recall in the early days this was even validated in depth, allowing certain path to be modified, others not. At least this is no longer the case. Still we encode misc state into this structure, all of it to be modifiable by users in UI and API. 

Removing in `censorUser` seems to be just wrong, as we might even rely on the updated state in the frontend.